sebsync.py

```diff
@@ -138,42 +151,84 @@ def get_remote_ebooks() -> None:
         click.echo(f"Found {len(remote_ebooks)} remote ebooks.")
 
 
-def get_local_ebooks() -> None:
+def get_local_ebooks(local_cache: dict) -> None:
     """Retrieve metadata of Standard EPUBs in the specified directory and subdirectories."""
-    for path in options.books.glob("**/*.epub"):
-        if not path.is_file():
-            continue
-        try:
-            with zipfile.ZipFile(path) as zip:
-                with zip.open("META-INF/container.xml") as file:
-                    root = ElementTree.parse(file)
-                    ns = {"container": "urn:oasis:names:tc:opendocument:xmlns:container"}
-                    rootfile = root.find(".//container:rootfile", ns).attrib["full-path"]
-                with zip.open(rootfile) as file:
-                    root = ElementTree.parse(file)
-                    ns = {
-                        "opf": "http://www.idpf.org/2007/opf",
-                        "dc": "http://purl.org/dc/elements/1.1/",
-                    }
-                    metadata = root.find("opf:metadata", ns)
-                    id = metadata.find("dc:identifier", ns)
-                    if id is None or "standardebooks.org" not in id.text:
-                        continue
-                    modified = metadata.find(".//opf:meta[@property='dcterms:modified']", ns)
+    if options.type == "kindle":
+        # Retrieve the id, title, path, modified date, and hexdigest from our cache file
+        for path in options.books.glob("**/*.azw3"):
+            if not path.is_file():
+                continue
+            try:
+                if path.name in local_cache:
                     local_ebook = LocalEbook(
-                        id=id.text,
-                        title=metadata.find(".//dc:title", ns).text,
+                        id=local_cache[path.name].get("id"),
+                        title=local_cache[path.name].get("title"),
                         path=path,
-                        modified=fromisoformat(modified.text),
+                        modified=fromisoformat(local_cache[path.name].get("modified")),
+                        hexdigest=local_cache[path.name].get("hexdigest"),
                     )
                     local_ebooks.append(local_ebook)
-        except Exception:
-            echo_status(path, Status.UNKNOWN)
+                else:
```

Comment:
OK, so I see here now the attempt to use the hash. It seems simpler to just do this and not even try to index by path name, which I would expect  in a majority of cases, will never match due to being moved or renamed.